### PR TITLE
Replace loss function in BERT_LOSS with SoftmaxCrossEntropyLoss.

### PR DESCRIPTION
**Pull request by codemzs**

### CODE REVIEW

**Comment by ytaous** [comment]
Do u have the code that modify/re-gen the onnx file to check in as well?

File: orttraining/orttraining/core/graph/loss_func/bert_loss.h [hunk 1]
```diff
@@ -17,6 +17,10 @@ struct BertLoss : public ILossFunction {
                                          GraphAugmenter::GraphDefs& graph_defs);
   static TypeProto* GetGatheredPredictionTypeProto(const NodeArg* prediction_arg,
                                                    GraphAugmenter::GraphDefs& graph_defs);
+  static TypeProto* GetGatheredPredictionTransposedTypeProto(const NodeArg* prediction_arg,
+                                                   GraphAugmenter::GraphDefs& graph_defs);
```
Comment (ytaous) [inline orttraining/orttraining/core/graph/loss_func/bert_loss.h:21]: nit: format

File: orttraining/orttraining/models/bert/main.cc [hunk 2]
```diff
@@ -593,7 +593,6 @@ void setup_training_params(BertParameters& params) {
       {"input_mask", "input3"},
       {"masked_lm_positions", "masked_lm_positions"},
       {"masked_lm_ids", "masked_lm_ids"},
```
Comment (ytaous) [inline orttraining/orttraining/models/bert/main.cc:595]: Would existing onnx models fail? for example, e2e/perf test pipelines? Perhaps take the original Bert toy model, and run it, if it works, then it's fine.

**Comment by ytaous** [comment]
>   const auto model_path = ORT_TSTR("testdata/bert_toy_optimized.onnx");

try the onnx before it's modified see if it's still backward compatible.

---
Refers to: orttraining/orttraining/test/graph/gradient_graph_builder_test.cc:463 in 3039c5a. [](commit_id = 3039c5ab90c0b10b56884fcd7a3e4963acd89dd6, deletion_comment = False)

File: orttraining/orttraining/models/bert/main.cc (hunk 2 above)
Comment (codemzs) [reply orttraining/orttraining/models/bert/main.cc:595]: There is a BERT tiny test that passes but I can run more tests, also will push another commit to sync with master because it seems more tests were added that use BERT-tiny e2e training

**Comment by codemzs** [comment]
> > const auto model_path = ORT_TSTR("testdata/bert_toy_optimized.onnx");
> 
> try the onnx before it's modified see if it's still backward compatible.
> 
> Refers to: orttraining/orttraining/test/graph/gradient_graph_builder_test.cc:463 in [3039c5a](https://github.com/microsoft/onnxruntime/commit/3039c5ab90c0b10b56884fcd7a3e4963acd89dd6). [](commit_id = [3039c5a](https://github.com/microsoft/onnxruntime/commit/3039c5ab90c0b10b56884fcd7a3e4963acd89dd6), deletion_comment = False)

Do we really care about backward compatibility? this is just for internal models right? may be I could be missing something, @SherlockNoMad ?

**Comment by ytaous** [comment]
if the the code can run against existing models, will be great.

---
In reply to: [659030620](https://github.com/microsoft/onnxruntime/pull/4509#issuecomment-659030620) [](ancestors = 659030620)

File: orttraining/orttraining/models/bert/main.cc (hunk 2 above)
Comment (ytaous) [reply orttraining/orttraining/models/bert/main.cc:595]: one is good enough :-)

---
In reply to: [455380227](https://github.com/microsoft/onnxruntime/pull/4509#discussion_r455380227) [](ancestors = 455380227)

**Comment by ytaous** [comment]
>   const auto model_path = ORT_TSTR("testdata/bert_toy_optimized.onnx");

since you've tested bert tiny, then we are good.

---
In reply to: [659030083](https://github.com/microsoft/onnxruntime/pull/4509#issuecomment-659030083) [](ancestors = 659030083)

---
Refers to: orttraining/orttraining/test/graph/gradient_graph_builder_test.cc:463 in 3039c5a. [](commit_id = 3039c5ab90c0b10b56884fcd7a3e4963acd89dd6, deletion_comment = False)

**Comment by codemzs** [comment]
> Do u have the code that modify/re-gen the onnx file to check in as well?

@ytaous I used your script: orttraining/tools/scripts/opset12_model_transform.py
